ibis/expr/api.py

```diff
@@ -2173,35 +2177,50 @@ def difference(table: ir.Table, *rest: ir.Table, distinct: bool = True) -> ir.Ta
     --------
     >>> import ibis
     >>> ibis.options.interactive = True
-    >>> t1 = ibis.memtable({"a": [1, 2]})
-    >>> t1
+    >>> t1 = ibis.memtable({"a": [7, 8, 8, 9, 9, 9]})
+    >>> t2 = ibis.memtable({"a": [8, 9]})
+
+    With distinct=True, if a row ever appears in any of `*rest`,
+    it will not appear in the result.
+    So here, all appearances of 8 and 9 are removed:
+
+    >>> t1.difference(t2)
     ┏━━━━━━━┓
     ┃ a     ┃
     ┡━━━━━━━┩
     │ int64 │
     ├───────┤
-    │     1 │
-    │     2 │
+    │     7 │
     └───────┘
-    >>> t2 = ibis.memtable({"a": [2, 3]})
-    >>> t2
+
+    With `distinct=False`, the algorithm is more of a multiset/bag difference.
```

Comment:
```suggestion
    With `distinct=False`, the algorithm is a [multiset](https://en.wikipedia.org/wiki/Multiset) difference.
```

Let's add a link to the wikipedia page for multiset in case the terminology or concept is unfamiliar to users.